WordPress/Classes/Utility/ZendeskUtils.swift

```diff
@@ -69,20 +71,35 @@ import CoreTelephony
             let presentInController = ZendeskUtils.configureViewController(controller)
             ZDKRequests.presentRequestCreation(with: presentInController)
             self.createRequest()
-        })
+        }
     }
 
 
     func showTicketListIfPossible(from controller: UIViewController) {
-        ZendeskUtils.createIdentity(completion: { success in
+        ZendeskUtils.createIdentity { success in
             guard success else {
                 // TODO: show error
                 return
             }
 
             let presentInController = ZendeskUtils.configureViewController(controller)
             ZDKRequests.presentRequestList(with: presentInController)
-        })
+        }
+    }
+
+    static func setNeedToRegisterDevice(_ identifier: String) {
+        ZendeskUtils.sharedInstance.needToRegisterDevice = true
+        ZendeskUtils.sharedInstance.deviceID = identifier
```

Comment:
Should we call `createZendeskIdentity` here so we can go ahead and create one if we have everything we need at this point?

Reply:
Well, the point of postponing device registration (and thus identity creation) is that at this point the user's account isn't fully loaded, so the identity won't be created anyway. When a user accesses a Support option, the identity is then created and the device registered if that's successful. Doing them in tandem works best. I don't see a reason to do it otherwise...

Reply:
If this isn't the first launch of the app (so the user is already logged in), then we'll have account details ready?

Reply:
Sure, probably. Maybe I'm not understanding, but I don't see what it gains us.  `createZendeskIdentity` is called when a ZD view is needed. 

Plus, now that I think about it, creating the identity this early may cause issues with self-hosted users. In this case, we save the user information used for Zendesk to User Defaults. If we create the identity, they change or add sites, then access Support, we've potentially saved the wrong information to UD, thus using the wrong information for Zendesk.

Reply:
Okay, sounds good – just wanted to check if we could do it sooner :) Totally makes sense!